src/models/message.ts

```diff
@@ -449,4 +451,65 @@ export class MessageModel extends CosmosdbModelTTL<
       TE.mapLeft(error => new Error(error.message))
     );
   }
+
+  /**
+   * Retrieve the message content from a blob with container client
+   *
+   * @param containerClient The azure.ContainerClient used to store the media
+   * @param messageId The id of the message used to set the blob name
+   */
+  public getContentFromContainer(
+    containerClient: ContainerClient,
+    messageId: string
+  ): TE.TaskEither<Error, Option<MessageContent>> {
+    // Retrieve blob content and deserialize
+    return pipe(
+      blobIdFromMessageId(messageId),
+      getBlobFromContainerAsTextWithError(containerClient),
+      TE.mapLeft(storageError => ({
+        code: storageError.code ?? GenericCode,
+        message: storageError.message
+      })),
+      TE.chain(maybeContentAsText =>
+        TE.fromEither(
+          E.fromOption(
+            // Blob exists but the content is empty
+            () => ({
+              code: GenericCode,
+              message: "Cannot get stored message content from empty blob"
+            })
+          )(maybeContentAsText)
+        )
+      ),
```

Comment:
I think we can simplify with

```suggestion
      TE.chain(
        TE.fromOption(
          // Blob exists but the content is empty
          () => ({
            code: GenericCode,
            message: "Cannot get stored message content from empty blob"
          })
        )
      ),
```

Reply:
This code is intended to be "identical" to the other one, at least on a first step. We need it to behave as much identically as possibile.